pkg/storage/bucket/client.go

```diff
@@ -39,17 +41,21 @@ const (
 
 	// Filesystem is the value for the filesystem storage backend.
 	Filesystem = "filesystem"
+
+	validPrefixCharactersRegex = `^[\da-zA-Z\-_.*'()!]+$`
 )
 
 var (
 	SupportedBackends = []string{S3, GCS, Azure, Swift, Filesystem}
 
-	ErrUnsupportedStorageBackend = errors.New("unsupported storage backend")
+	ErrUnsupportedStorageBackend        = errors.New("unsupported storage backend")
+	ErrInvalidCharactersInStoragePrefix = errors.New("storage_prefix contains invalid characters, it should match the Go regexp " + validPrefixCharactersRegex)
```

Comment:
Can we be explicit about which characters are allowed? Message saying "it should match this regexp" may not be helpful for many people :(

Reply:
My consideration was that the regexp may change with time, but the help text or the error may be forgotten. 

Pushed a change in 7d74f12435905f37d871bb2d21a1f6ac45d3f10f